toolchain/check/testdata/alias/no_prelude/fail_params.carbon

```diff
@@ -12,7 +12,7 @@
 // CHECK:STDERR: alias A(T:! type) = T*;
 // CHECK:STDERR:        ^~~~~~~~~~
 // CHECK:STDERR:
-// CHECK:STDERR: fail_params.carbon:[[@LINE+7]]:21: error: `T` read before initialization. [ReadDuringInitialization]
+// CHECK:STDERR: fail_params.carbon:[[@LINE+7]]:21: error: `T` read before initialization [ReadDuringInitialization]
```

Comment:
Please add a TODO to not produce this second error.

Reply:
Why? It seems like a valid error, and not directly downstream of the other errors. e.g. if I fix the first and third errors by making this a `let` instead of an `alias`, this error is still applicable, so it could be helpful to produce all three errors in a single build.

Reply:
`T` is bound to a parameter, so it would be initialized before any use on the right-hand side of the `=`. I didn't understand what you meant about `let` here -- `let` also can't be parameterized. But if we change this to be a class or function instead, we don't get this error:
```
class C(T:! type) {
  // OK
  var x: T*;
}
// OK
fn F(T:! type) -> T*;
// T used before initialization?
alias A(T:! type) = T*;
```

I guess the problem is that we never transition alias parameters out of the tombstone state (because we don't expect them to have parameters at all)?

Reply:
> `T` is bound to a parameter, so it would be initialized before any use on the right-hand side of the `=`.

Oh, of course! In other words, even if we allowed parameterized aliases, the initializer for an alias doesn't initialize the pattern. That being the case, we shouldn't be calling `StartPatternInitializer` and `EndPatternInitializer` when handling an alias initializer. I've fixed that now, so no need for a TODO.